File: frontend/src/types/conversation.ts
export interface Conversation {
  id: string;
  title: string;
  model: string;
  provider: string;
  space_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface ConversationWithMessages extends Conversation {
  messages: {
    id: string;
    role: string;
    content: string;
    tool_calls: any;
    files: any;
    created_at: string;
  }[];
}

File: backend/app/models/conversation.py
import uuid
from datetime import datetime
from sqlalchemy import String, DateTime, ForeignKey, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.database import Base


class Conversation(Base):
    __tablename__ = "conversations"

    id: Mapped[str] = mapped_column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id: Mapped[str] = mapped_column(String, ForeignKey("users.id"), index=True)
    title: Mapped[str] = mapped_column(String(500), default="New Conversation")
    model: Mapped[str] = mapped_column(String(100), default="gpt-4o")
    provider: Mapped[str] = mapped_column(String(50), default="openai")
    space_id: Mapped[str | None] = mapped_column(String, ForeignKey("spaces.id"), nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
    updated_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    user = relationship("User", back_populates="conversations")
    messages = relationship("Message", back_populates="conversation", cascade="all, delete-orphan", order_by="Message.created_at")
    space = relationship("Space", back_populates="conversations")
